### PR TITLE
Upgrade PgBouncer to v1.16.0 [standalone]

### PR DESCRIPTION
**Pull request by atsikham**
Task #2453

### CODE REVIEW

**Comment by plirglo** [comment]
Changelog update is missing

**Review by erzetpe — DISMISSED** [review]

**Comment by atsikham** [comment]
> Changelog update is missing

@plirglo Updated

**Review by erzetpe — DISMISSED** [review]

File: CHANGELOG-1.2.md [hunk 1]
```diff
@@ -25,8 +25,9 @@
 - [#2076](https://github.com/epiphany-platform/epiphany/issues/2076) - Upgrade of PgBouncer [Kubernetes] to v1.15.0
 - [#1797](https://github.com/epiphany-platform/epiphany/issues/1797) - Upgrade Keycloak to v14.0.0
 - [#1861](https://github.com/epiphany-platform/epiphany/issues/1861) - Upgrade PostgreSQL to v13
-- [#2074](https://github.com/epiphany-platform/epiphany/issues/2074) - Upgrade repmgr to 5.2.1
-- [#2077](https://github.com/epiphany-platform/epiphany/issues/2077) - Upgrade PgAudit to 1.5.0
+- [#2074](https://github.com/epiphany-platform/epiphany/issues/2074) - Upgrade repmgr to v5.2.1
+- [#2077](https://github.com/epiphany-platform/epiphany/issues/2077) - Upgrade PgAudit to v1.5.0
+- [#2453](https://github.com/epiphany-platform/epiphany/issues/2453) - Upgrade of PgBouncer to v1.16.0 [standalone]
```
Comment (to-bar) [inline CHANGELOG-1.2.md]:
```suggestion
- [#2453](https://github.com/epiphany-platform/epiphany/issues/2453) - Upgrade PgBouncer to v1.16.0 [standalone]
```

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/upgrade/upgrade.yml [hunk 2]
```diff
@@ -6,7 +6,8 @@
 - name: PostgreSQL | Set facts on installed extensions
   set_fact:
     pg_is_pgaudit_used: "{{ ansible_facts.packages.keys() | intersect(_packages.pgaudit) | count > 0 }}"
-    pg_is_repmgr_used:  "{{ ansible_facts.packages.keys() | intersect(_packages.repmgr)  | count > 0 }}"
+    pg_is_repmgr_used: "{{ ansible_facts.packages.keys() | intersect(_packages.repmgr)  | count > 0 }}"
+    pg_is_pgbouncer_used: ansible_facts.packages.pgbouncer is defined # package name is the same for all versions
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/upgrade/upgrade.yml]:
```suggestion
    pg_is_pgbouncer_used: "{{ ansible_facts.packages.pgbouncer is defined }}" # package name is the same for all versions
```

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/pgbouncer.ini.j2 [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+
+;; fast, normal, secure, legacy, <ciphersuite string>
+;server_tls_ciphers = fast
+
+;;;
+;;; Authentication settings
+;;;
+
+;; any, trust, plain, md5, cert, hba, pam
+auth_type = trust
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/pgbouncer.ini.j2]: I'm not sure if `auth_type = trust` is a good idea.

> Also, make sure auth_type is not set to trust in pgbouncer.ini – instead, you should set trust in pg_hba.conf for auth_user and clamp it down to only the IP(s) that will be running PgBouncer. 

source: https://www.enterprisedb.com/postgres-tutorials/pgbouncer-authquery-and-authuser-pro-tips

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/userlist.txt.j2 [hunk 4]
```diff
@@ -0,0 +1 @@
+"postgres" "*"
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/userlist.txt.j2]: This will overwrite this file on each `epicli apply`.

**Review by to-bar — CHANGES_REQUESTED** [review]

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/extensions/pgbouncer/extension.yml [hunk 5]
```diff
@@ -1,38 +1,44 @@
...
+        state: started
 
-    - name: Extensions | PgBouncer | Change pgbouncer users configuration
-      lineinfile:
-        path: /etc/pgbouncer/userlist.txt
-        line: '"postgres" "*"'
-        create: true
+    - name: Extensions | PgBouncer | Copy configuration files
+      vars:
+        _files:
```
Comment (plirglo) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/extensions/pgbouncer/extension.yml]: I think we shouldn't overwrite existing configuration if we re-run apply. I would stay with previous configuration and just replace particular lines.

**Review by plirglo — COMMENTED** [review]
We should consider to add information that this version (standalone) will be deprecated and we should get rid of this component in this state in future releases since i think no one is using it. On production cluster at least two postgresql vms should be used to ensure high availability. This configuration was more useful when native postgresql replication was supported by Epiphany.

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/extensions/pgbouncer/extension.yml (hunk 5 above)
Comment (atsikham) [reply core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/extensions/pgbouncer/extension.yml]: I don’t agree. It’s done the same as in all other places. IMO everything we support should be configured by epiphany, not manually.

If we need to have an ability to modify this file, it should be done by epiphany, not manually.

Also this was done following this comment - https://github.com/epiphany-platform/epiphany/pull/1845#pullrequestreview-531363344

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/pgbouncer.ini.j2 (hunk 3 above)
Comment (atsikham) [reply core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/pgbouncer.ini.j2]: With using lineinfile again we don't control this setting.

**Review by erzetpe — DISMISSED** [review]

**Review by to-bar — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Pull request merged**